src/Sulu/Bundle/WebsiteBundle/Controller/DefaultController.php

```diff
@@ -76,13 +76,23 @@ protected function resolveRedirectUrl($redirectUrl, $requestUri)
         $redirectInfo = $this->parseUrl($redirectUrl);
         $requestInfo = $this->parseUrl($requestUri);
 
-        $url = sprintf('%s://%s', $requestInfo['scheme'], $redirectInfo['host']);
+        $url = sprintf('%s://%s', $requestInfo['scheme'], $requestInfo['host']);
+
+        if (isset($redirectInfo['host'])) {
+            $url = sprintf('%s://%s', $requestInfo['scheme'], $redirectInfo['host']);
+        }
 
         if (isset($requestInfo['port'])) {
             $url .= ':' . $requestInfo['port'];
         }
 
-        if (isset($redirectInfo['path'])) {
+        if (
+            isset($redirectInfo['path'])
+            && (
+                // if requested url not starting with redirectUrl it need to be added
+                !isset($requestInfo['path'])
+                || strpos($requestInfo['path'], $redirectInfo['path'] . '/') !== 0
+        )) {
```

Comment:
if the redirect path is `/de` and the requested path is `/de/articles/` => it will be `/de/articles`.
if the redirect path is `/de` and the requested path is `/articles/` => it will be correctly `/de/articles`

which case you miss in the `DefaultControllerTest`?